src/monitor-method-workspace/monitor-method.repository.ts

```diff
@@ -10,12 +10,15 @@ export class MonitorMethodWorkspaceRepository extends Repository<
     locationId: string,
     parameterCode: string,
     beginDate: Date,
+    endDate: Date,
+
   ): Promise<MonitorMethod> {
     return this.findOne({
       where: {
         locationId,
         parameterCode,
         beginDate,
+        endDate,
```

Comment:
Same fix needed as for Duct Waf.  Include endDate in where clause only if it is not null, and OR'd to the beginDate comparison